src/map/mob.cpp

```diff
@@ -3399,8 +3405,14 @@ void mob_heal(struct mob_data *md,unsigned int heal)
 		for(i = 0; i < DAMAGELOG_SIZE; i++)// must show hp bar to all char who already hit the mob.
 			if( md->dmglog[i].id ) {
 				struct map_session_data *sd = map_charid2sd(md->dmglog[i].id);
-				if( sd && check_distance_bl(&md->bl, &sd->bl, AREA_SIZE) ) // check if in range
```

Comment:
Needs additional call above the loop to update md->master.
Master will never be inside damage log and also needs HP update.